### DISCUSSION TITLE
Open a tag using Markup() but close it later (question)

### DISCUSSION THREAD

**Discussion by daningalla** [spectreconsole/spectre.console#506] [Ideas]
I'm writing a customizable MSFT Logger implementation for SpectreConsole. I need the ability to open a markup tag in order to apply styling to the event output, render the event, then close the tag after event rendering has been completed. My implementation has a whole pipeline around this. The styling per log level is customizable by the consumer, so these operations need to be isolated. I'd prefer not to have to build a string, my pipeline puts rendering fragments to the console bit by bit until the log event is rendered in full.

When I call the Markup() method it throws an exception because a styling tag isn't closed, even though I have the intention of closing it later. Is there any way to write incomplete markup to the call?

For now, I'll let the user customize just the foreground color, but it would be nice to be to define a complete markup (e.g. [blue on yellow]) in a string vs having to pick a color.

**Comment by patriksvensson** [comment]
I'm not against adding support for non-terminated markup tags, but with great powers come great responsibility. There won't be any guarantees that a markup tag is closed and the output style being restored. I will need to think about this a bit and discuss it with @phil-scott-78.

**Comment by daningalla** [comment]
Thank you kindly for the reply @patriksvensson  - I've put `StringBuilder` in to buffer my rendering pipeline so I've at least solved my own challenge. I was just making sure I wasn't missing something in the current API. Without digging, I'm sure there's a solution hiding in there with `IRenderable`.

**Comment by patriksvensson** [comment]
We've been thinking about this and concluded that this is not something that we would want to support until we've shipped version 1.0 of Spectre.Console. Thanks for opening the issue, though!
